res/controllers/common-controller-scripts.js

```diff
@@ -109,561 +102,463 @@ var msecondstominutes = function(msecs) {
+                in the specified direction (positive is forwards and is the default)
+                If the current loop length is shorter than the requested move distance,
+                it's only moved a distance equal to its length.
+       Input:   MixxxControl group, direction to move, number of beats to move
+       Output:  none
+       -------- ------------------------------------------------------ */
+    loopMove: function(group, direction, numberOfBeats) {
+        if (!numberOfBeats || numberOfBeats === 0) { numberOfBeats = 0.5; }
+
+        if (direction < 0) {
+            engine.setValue(group, "loop_move", -numberOfBeats);
+        } else {
+            engine.setValue(group, "loop_move", numberOfBeats);
+        }
+    },
+
+    /* -------- ------------------------------------------------------
+         script.midiPitch
+       Purpose: Takes the value from a little-endian 14-bit MIDI pitch
+                wheel message and returns the value for a "rate" (pitch
+                slider) Mixxx control
+       Input:   Least significant byte, most sig. byte, MIDI status byte
+       Output:  Value for a "rate" control, or false if the input MIDI
+                message was not a Pitch message (0xE#)
+       -------- ------------------------------------------------------ */
+    // TODO: Is this still useful now that MidiController.cpp properly handles these?
+    midiPitch: function(LSB, MSB, status) {
+        if ((status & 0xF0) !== 0xE0) { // Mask the upper nybble so we can check the opcode regardless of the channel
+            console.log(`Script.midiPitch: Error, not a MIDI pitch (0xEn) message: ${  status}`);
+            return false;
+        }
+        const value = (MSB << 7) | LSB; // Construct the 14-bit number
+
+        // Range is 0x0000..0x3FFF center @ 0x2000, i.e. 0..16383 center @ 8192
+        const rate = (value - 8192) / 8191;
+        //     print("Script.Pitch: MSB="+MSB+", LSB="+LSB+", value="+value+", rate="+rate);
+        return rate;
+    },
+    /* -------- ------------------------------------------------------
+         script.spinback
+       Purpose: wrapper around engine.spinback() that can be directly mapped
+                from xml for a spinback effect
+                e.g: <key>script.spinback</key>
+       Input:   channel, control, value, status, group, factor (optional), start rate (optional)
+       Output:  none
+       -------- ------------------------------------------------------ */
+    spinback: function(channel, control, value, status, group, factor, rate) {
+        // if brake is called without defined factor and rate, reset to defaults
+        if (factor === undefined) {
+            factor = 1;
+        }
+        // if brake is called without defined rate, reset to default
+        if (rate === undefined) {
+            rate = -10;
+        }
+        // disable on note-off or zero value note/cc
```

Comment:
does our version of js support default parameters?

```suggestion
    spinback: function(channel, control, value, status, group, factor = 1, rate = -10) {
        // disable on note-off or zero value note/cc
```